### PR TITLE
Backports + release notes for v3.4.0 final

### PR DESCRIPTION
**Pull request by mheon**
Does not include release commit right now, as I'm working with @baude to get one final commit landed. We can merge regardless, that can be in a separate commit if necessary.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
especially if quay.io starts to cooperate, lots of red tests at the moment.

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -68,6 +71,12 @@
...
 - Fixed a bug where the `podman run` and `podman start --attach` commands could race when retrieving the exit code of a container that had already been removed resulting in an error (e.g. by an external `podman rm -f`) ([#11633](https://github.com/containers/podman/issues/11633)).
 - Fixed a bug where the `podman generate kube` command would add default environment variables to generated YAML.
+- Fixed a bug where the `podman generate kube` command would add the default CMD from the image to generated YAML ([#11672](https://github.com/containers/podman/issues/11672)).
+- Fixed a bug where the `podman rm --storage` command could fail to remove containers under some circumstances ([#11207](https://github.com/containers/podman/issues/11207)).
+- Fixed a bug where the CNI plugins were unable to locate state information when Podman was run as root, leading to a failure to properly tear down container networking ([#10745](https://github.com/containers/podman/issues/10745) and [#10806](https://github.com/containers/podman/issues/10806)).
```
Comment (Luap99) [inline RELEASE_NOTES.md:76]: Please remove this, it is not fixed in 3.4

Comment (mheon) [reply RELEASE_NOTES.md:76]: Oh, I thought that the revert also fixed this by unsetting XDG_RUNTIME_DIR for the CNI plugins. Removing, then...

Comment (Luap99) [reply RELEASE_NOTES.md]: Yes but only for 4.0, the 3.4 code does not use the network interface

**Review by Luap99 — COMMENTED** [review]
LGTM

**Comment by mheon** [comment]
@containers/podman-maintainers Merge please, I can handle the release commit and the final backport in a separate PR.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Pull request merged**